src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -713,7 +713,8 @@ abstract class BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
               case StringTag  =>
                 assert(const.value != null, const) // TODO this invariant isn't documented in `case class Constant`
                 av.visit(name, const.stringValue)  // `stringValue` special-cases null, but that execution path isn't exercised for a const with StringTag
-              case ClazzTag   => av.visit(name, typeToBType(const.typeValue).toASMType)
```

Comment:
It turns out that code works fine. The reason is that erasure visits the `Literal(Constant(..))` AST and replaces the type by a `ClassNoArgsTypeRef` to `List`. In the case of an annotation arg, the `Constant(..)` is part of a symbol's `annotations` and not touched by erasure.